### PR TITLE
User authentication completed

### PR DESCRIPTION
**Pull request by elbrunof**

### CODE REVIEW

File: test/models/comment_test.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class CommentTest < ActiveSupport::TestCase
```
Comment (franciscoaguirre) [inline test/models/comment_test.rb]: Please remove scaffolding.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -7,9 +7,16 @@
...
   <body>
+    <% if notice %>
+      <p class="alert alert-success"><%= notice %></p>
+    <% end %>
+    <% if alert %>
+      <p class="alert alert-danger"><%= alert %></p>
+    <% end %>
     <%= yield %>
-  </body>
+  </body> <div></div>
```
Comment (franciscoaguirre) [inline app/views/layouts/application.html.erb]: What is this `<div>` doing here?

File: app/assets/stylesheets/layout.scss [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+}
+
+.modal .box {
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+  text-align: center;
+  background-color: #FFF;
+  color: #A5CD39;
```
Comment (franciscoaguirre) [inline app/assets/stylesheets/layout.scss]: Order css properties alphabetically.

File: Gemfile [hunk 4]
```diff
@@ -22,6 +22,14 @@ gem 'coffee-rails', '~> 4.2'
 gem 'turbolinks', '~> 5'
 # Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
 gem 'jbuilder', '~> 2.5'
+
+
+#Use Postgresql as database
+gem 'pg'
```
Comment (santiago-bertinat) [inline Gemfile]: lets specify the gem version

File: Gemfile [hunk 5]
```diff
@@ -22,6 +22,14 @@ gem 'coffee-rails', '~> 4.2'
 gem 'turbolinks', '~> 5'
 # Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
 gem 'jbuilder', '~> 2.5'
+
+
+#Use Postgresql as database
```
Comment (santiago-bertinat) [inline Gemfile]: you need to leave a space after #

File: app/assets/stylesheets/form.scss [hunk 6]
```diff
@@ -0,0 +1,43 @@
+.primary_form {
+  margin: 20px 0;
+
+  label {
+    display: block;
+    margin: 10px 0 0 0;
+  }
+
+  em,i {
```
Comment (santiago-bertinat) [inline app/assets/stylesheets/form.scss]: space after ,

File: app/assets/stylesheets/form.scss [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    height: 40px;
+    padding: 10px;
+    width: 250px;
+  }
+
+  .not_block {
+    display: inline;
+  }
+}
+.btn {
```
Comment (santiago-bertinat) [inline app/assets/stylesheets/form.scss]: new line between css declarations

File: app/assets/stylesheets/layout.scss [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+  font-family: 'Montserrat';
+  color: #FFF;
+  background-color: #00ADEE;
+  background-image: url(images/background.png);
+}
+
+.mt-10 {
+  margin-top: 10px;
+}
+.wrapper {
```
Comment (santiago-bertinat) [inline app/assets/stylesheets/layout.scss]: new line

File: app/assets/stylesheets/layout.scss [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+  color: #A5CD39;
+}
+
+.modal .box span {
+  display: inline-block;
+  padding: 35px 25px;
+  font-size: 32px;
+}
+
+.bg-image {
```
Comment (santiago-bertinat) [inline app/assets/stylesheets/layout.scss]: when do you need this?

File: app/assets/stylesheets/layout.scss [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+  position: absolute;
+  z-index: -1;
+  height: 100%;
+  width: 100%;
+}
+
+h1,h2 {
```
Comment (santiago-bertinat) [inline app/assets/stylesheets/layout.scss]: space after ,

File: app/controllers/users/confirmations_controller.rb [hunk 11]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Users::ConfirmationsController < Devise::ConfirmationsController
```
Comment (santiago-bertinat) [inline app/controllers/users/confirmations_controller.rb]: Remove, not necessary

File: app/controllers/users/unlocks_controller.rb [hunk 12]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+class Users::UnlocksController < Devise::UnlocksController
```
Comment (santiago-bertinat) [inline app/controllers/users/unlocks_controller.rb]: Remove all devise controllers, they are not necessary unless you need to override them

File: app/models/comment.rb [hunk 13]
```diff
@@ -0,0 +1,3 @@
+class Comment < ApplicationRecord
+  belongs_to :task
```
Comment (santiago-bertinat) [inline app/models/comment.rb]: please lets add validations

File: app/models/task.rb [hunk 14]
```diff
@@ -0,0 +1,3 @@
+class Task < ApplicationRecord
+  has_many :comments
```
Comment (santiago-bertinat) [inline app/models/task.rb]: please lets add validations for all models

File: app/views/devise/confirmations/new.html.erb [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+<%= form_for(resource, as: resource_name, url: confirmation_path(resource_name), html: { method: :post }) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email", value: (resource.pending_reconfirmation? ? resource.unconfirmed_email : resource.email) %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Resend confirmation instructions" %>
```
Comment (santiago-bertinat) [inline app/views/devise/confirmations/new.html.erb]: Always single quotes if you are on ruby. For HTML double quotes.
please update all views

File: app/views/devise/mailer/email_changed.html.erb [hunk 16]
```diff
@@ -0,0 +1,7 @@
+<p>Hello <%= @email %>!</p>
```
Comment (santiago-bertinat) [inline app/views/devise/mailer/email_changed.html.erb]: Remove all views that we are not overriding

File: test/fixtures/comments.yml [hunk 17]
```diff
@@ -0,0 +1,9 @@
+# Read about fixtures at http://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (santiago-bertinat) [inline test/fixtures/comments.yml]: Remove everything on test, you can even delete the directory.
We are going to use rspec for the test suite

File: app/assets/stylesheets/form.scss [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+  text-align: center;
+  text-decoration: none;
+  text-transform: uppercase;
+}
+.primary_btn {
+  background-color: transparent;
+  border: solid 2px #FFF;
+}
+.danger_btn {
+  background-color: #e74c3c;
```
Comment (guillermoap) [inline app/assets/stylesheets/form.scss]: pls use scss variables for colour palette

File: app/views/index/index.erb [hunk 19]
```diff
@@ -18,24 +18,25 @@
     <h1>Organize your monthly tasks</h1>
     <p>Create and control your monthly tasks to never miss one. You can mark your tasks as completed. If you are reaching end of the month and one is not completed, we will send you a reminder! Each month the task will automatically get back to un completed</p>
     <ul id="task-list">
-      <%
+      <%#
```
Comment (guillermoap) [inline app/views/index/index.erb]: Remove commented code

File: config/database.yml [hunk 20]
```diff
@@ -5,13 +5,13 @@
 #   gem 'sqlite3'
```
Comment (guillermoap) [inline config/database.yml]: add the `database.yml` to gitignore

File: config/database.yml [hunk 21]
```diff
@@ -5,13 +5,13 @@
...
 #
 default: &default
-  adapter: sqlite3
-  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
-  timeout: 5000
+  adapter: postgresql
+  encoding: unicode
```
Comment (guillermoap) [inline config/database.yml]: use `utf-8` and add username

**Comment by guillermoap** [comment]
Regarding the commits for the next changes divide them by feat/goal/feature. So instead of having one big commit you have several more specific ones.
Also give them more descriptive names, so that when i read the commit I can more or less know what those changes are.

**Pull request merged**